### PR TITLE
Also initialize the per-module SwiftASTContext using ProcessModule()

### PR DESCRIPTION
**Pull request by adrian-prantl**
this resolves some inconsistencies in search path resolution between
the expression context and SwiftASTContextForModule().

rdar://127776675

### CODE REVIEW

**Comment by adrian-prantl** [comment]
@swift-ci test

File: lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp [hunk 1]
```diff
@@ -1990,6 +1990,156 @@ static bool IsSerializedAST(const swift::ModuleDecl &module) {
   });
 }
 
+/// Scan a newly added lldb::Module for Swift modules and report any errors in
+/// its module SwiftASTContext to Target.
+static void
+ProcessModule(Module &module, std::string m_description,
```
Comment (augusto2112) [inline lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp:1996]: Nit: rename this from `m_description` to `description` or make this a member function and not pass it as a parameter at all.

File: lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp [hunk 2]
```diff
@@ -5469,11 +5470,15 @@ void SwiftASTContextForExpressions::ModulesDidLoad(ModuleList &module_list) {
     std::vector<std::pair<std::string, bool>> framework_search_paths;
     std::vector<std::string> extra_clang_args;
     lldb::ModuleSP module_sp = module_list.GetModuleAtIndex(mi);
+    if (!module_sp)
```
Comment (augusto2112) [inline lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp:5474]: Should this be an assert instead? The module should never be null if iterating over the module list.

File: lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp [hunk 3]
```diff
@@ -2170,11 +2320,22 @@ SwiftASTContext::CreateInstance(lldb::LanguageType language, Module &module,
 
   swift_ast_sp->SetCompilerInvocationLLDBOverrides();
 
-  // Apply the working directory to all relative paths.
-  std::vector<std::string> DeserializedArgs = swift_ast_sp->GetClangArguments();
+  // Collect search paths before importing modules.
+  const bool discover_implicit_search_paths = false;
+  const bool use_all_compiler_flags = false;
```
Comment (augusto2112) [inline lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp:2325]: If you're declaring variables for these parameters you could also declare one for `is_main_executable`

File: lldb/test/API/lang/swift/framework_paths/TestSwiftFrameworkPaths.py [hunk 4]
```diff
@@ -20,13 +16,25 @@ def test_system_framework(self):
             self, 'break here', lldb.SBFileSpec('main.swift'))
 
         log = self.getBuildArtifact("types.log")
-        self.runCmd('log enable lldb types -f "%s"' % log)
+        self.expect('log enable lldb types -f "%s"' % log)
```
Comment (augusto2112) [inline lldb/test/API/lang/swift/framework_paths/TestSwiftFrameworkPaths.py:21]: Why did you change this to `expect`?

**Review by augusto2112 — APPROVED** [review]

File: lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp (hunk 1 above)
Comment (adrian-prantl) [reply lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp:1996]: No, this is a hidden dependency of the LLDB_PRINTF() macro in this file.

File: lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp (hunk 2 above)
Comment (adrian-prantl) [reply lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp:5474]: A shared pointer to a [Module](https://lldb.llvm.org/cpp_reference/classlldb__private_1_1Module.html) which can contain NULL if idx is out of range.

Comment (adrian-prantl) [reply lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp:5474]: Since we are not using ForEach here, I'm not sure if other threads aren't allowed to modifiy the list while we're iterating.

File: lldb/test/API/lang/swift/framework_paths/TestSwiftFrameworkPaths.py (hunk 4 above)
Comment (adrian-prantl) [reply lldb/test/API/lang/swift/framework_paths/TestSwiftFrameworkPaths.py:21]: I thought that self.runCmd doesn't fail if the command fails. But I may be wrong about this.

**Comment by adrian-prantl** [comment]
@swift-ci test

**Comment by adrian-prantl** [comment]
@swift-ci test

**Comment by adrian-prantl** [comment]
@swift-ci test linux

**Comment by adrian-prantl** [comment]
test with https://github.com/apple/swift/pull/73757

**Comment by adrian-prantl** [comment]
test with https://github.com/apple/swift/pull/73757
@swift-ci test linux

**Comment by adrian-prantl** [comment]
@swift-ci test

**Comment by adrian-prantl** [comment]
@swift-ci test
